app/services/onetime/backfill_payment_option_installment_snapshots.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Onetime
+  class BackfillPaymentOptionInstallmentSnapshots
+    def self.perform
+      PaymentOption.where.not(product_installment_plan_id: nil)
+                   .left_joins(:installment_plan_snapshot)
+                   .where(installment_plan_snapshots: { id: nil })
+                   .find_in_batches(batch_size: 1000) do |batch|
+        batch.each do |payment_option|
```

Comment:
I think you can simplify. Can you check?
```suggestion
                   .find_each do |payment_option| # default batch size is 1000
```